addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -499,6 +499,289 @@ registry.slider = publicWidget.Widget.extend({
+            stopMovieOnBlur: false, autoPlay: params['autoplay'], showYTLogo: false, opacity: opacity, showControls: false,
+        };
+        var $el = $('<div/>', {'class': 'player', 'data-property': JSON.stringify(playerParams)});
+        var $loader = $("<span class='yt-loader'><span/></span>");
+        $videoContainer.append($el).append($loader);
+
+        var interval = null;
+        if ($("#oe_main_menu_navbar").length > 0) {
+            $loader.css("top", $("#oe_main_menu_navbar").outerHeight()+1);
+        }
+        $loader.animate({width: "45%"}, 800, function () {
+            var el = $loader;
+            interval = setInterval(function () { timer(); }, 300);
+            function timer() { var w =  el.width(); el.width(w + 5); }
+        });
+
+        if (!params['autoplay']) {
+            $el.one('YTPStart', function () {
+                $el.YTPPause();
+            });
+        }
+
+        var def = new Promise(function (resolve, reject) {
+            $el.on('YTPReady', function () {
+                clearInterval(interval);
+                $loader.css("width", "100%").fadeOut(500);
+
+                resolve($videoContainer.find('iframe'));
+
+                if (!params['controls'] && params['autoplay']) {
+                    return;
+                }
+
+                var $controls = $("<span/>", {'class': 'controls'}).appendTo($videoContainer);
+
+                var $btnplay = $("<span/>", {'class': 'fa fa-fw'}).appendTo($controls);
+                var playing = params['autoplay'];
+                $btnplay.toggleClass("fa-pause", playing).toggleClass("fa-play", !playing);
+                $btnplay.on("click", playCallback);
+                if (!params['controls']) {
+                    $btnplay.one('click', function () {
+                        $controls.remove();
+                    });
+                }
+
+                if (!params['muted'] && params['controls']) {
+                    var $btnMute = $("<span/>", {'class': 'fa fa-fw fa-volume-up'}).appendTo($controls);
+                    $btnMute.on("click", muteCallback);
+                }
+
+                function playCallback() {
+                    if (playing) {
+                        $el.YTPPause();
+                    } else {
+                        $el.YTPPlay();
+                    }
+                    playing = !playing;
+
+                    $btnplay.toggleClass("fa-pause", playing).toggleClass("fa-play", !playing);
+                }
+                function muteCallback() {
+                    $el.YTPToggleVolume();
+                    $btnMute.toggleClass("fa-volume-up").toggleClass("fa-volume-off");
+                }
+            });
+
+            $container.append($videoContainer);
+            $el.YTPlayer();
+        });
+        return def;
+    },
+
+    /**
+     * @private
+     */
+    _createYoutubeVideo: function () {
+        return this._ytPlayerVideo.apply(this, arguments);
```

Comment:
Why do you need two function which does the same thing ? Keep only `_createYoutubeVideo`